modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/OtelHelper.java

```diff
@@ -41,4 +48,36 @@ static Attributes fromMap(Map<String, Object> attributes) {
         });
         return builder.build();
     }
+
+    static Consumer<ObservableDoubleMeasurement> doubleMeasurementCallback(Supplier<DoubleWithAttributes> observer) {
+        return measurement -> {
+            DoubleWithAttributes observation;
+            try {
+                observation = observer.get();
+            } catch (RuntimeException err) {
+                assert false : "observer must not throw [" + err.getMessage() + "]";
```

Comment:
I see this code was just moved. Though, I'm a bit concerned about silently swallowing runtime exceptions if assertions are not explicitly enabled. That could  make it difficult to troubleshoot issues when metrics are missing.

Reply:
We could add logging here.

Reply:
That should help, I think that's worth doing 👍